extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -78,6 +81,20 @@ export class AnalysesResultsManager {
     return [...this.analysesResults];
   }
 
+  /**
+   * Prepares a directory for storing analysis results for a single query run.
+   * This directory initially contains only a timestamp file, which will be
+   * used by the query history manager to determine when the directory
+   * should be deleted.
+   *
+   * @param queryName The name of the query that was run.
+   */
+  public async prepareDownloadDirectory(queryName: string): Promise<void> {
```

Comment:
Do we only want analyses results to be stored or do we also want to store other artifacts and query information in this directory? 

If it's more general, then I think it belongs in the `RemoteQueriesManager` class.

Reply:
What else will need to be stored? Currently, it's only the analyses results. You probably have a better understanding here than I do.

Reply:
I reckon we'll need to be storing other information like actions workflow run id etc. We'll probably want to store the whole `RemoteQuery` object.